api/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -54,11 +54,11 @@ type IngressNodeFirewallProtocolRule struct {
 
 	// IngressNodeFirewallProtoRule define ingress node firewall rule for TCP, UDP and SCTP protocols.
 	// +optional
-	ProtocolRule IngressNodeFirewallProtoRule `json:"protoRule"`
+	ProtocolRule *IngressNodeFirewallProtoRule `json:"protoRule"`
 
 	// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols.
 	// +optional
-	ICMPRule IngressNodeFirewallICMPRule `json:"icmpRule"`
+	ICMPRule *IngressNodeFirewallICMPRule `json:"icmpRule"`
```

Comment:
why we need those as pointers fields ?

Reply:
I need a way to know if they are filling in by the user or just blank. `ICMPRule` field will be null if the user didnt define it.